dynamyc_list/list.cpp

```diff
@@ -0,0 +1,417 @@
+#include "list.h"
+
+/* Initializes list */
+void StringListInit(char*** list)
+{
+	if (*list)
+	{
+		StringListDestroy(list);
+	}
+	char** list_head = (char**)malloc(2 * sizeof(char*));
+
+	*list = list_head;//set pointer to head
+
+	list_head[0] = NULL;
+	list_head[1] = NULL;
+}
+
+/* Destroy list and set pointer to NULL. */
+void StringListDestroy(char*** list)
+{
+	char** current_elem = *list;
+	while (current_elem != NULL)
+	{
+		char** next_elem = (char**)current_elem[1];//get next element
+
+		free(current_elem[0]);//delete string in current_element
+		current_elem[0] = NULL;
+		free(current_elem);//delete current element
+		current_elem = NULL;
+
+		current_elem = next_elem;//go to next element
+	}
+	*list = NULL;
+}
+
+
+/* Inserts value at the end of the list. */
+void StringListAdd(char** list, String str)
+{
+	char** current_elem = list;
+	char** new_elem = (char**)malloc(2 * sizeof(char*));//create new elem
+	char* current_str = NULL;
+	current_str = (char*)malloc((strlen(str) + 1) * sizeof(char));
```

Comment:
Line 43 Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected)

Reply:
strlen() return string's size without \0, but for free() \0 naad at the string's end for correctly deletet it
Because it i add 1 to strlen() for \0


> Line 43 Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected)